docs/sources/reference/builder.md

```diff
@@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
 
 The `ENV` instruction sets the environment variable `<key>` to the value
 `<value>`. This value will be passed to all future 
-`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
-functionally equivalent to prefixing the command with `<key>=<value>`
+`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
+command without affecting future commands or containers, prefix the command
+with `<key>=<value>`.
```

Comment:
@erikh I'm not sure how precise you want me to be. IIRC the `RUN` command uses something like `/bin/sh -c` (assuming you're not using exec-style braces), right? So `RUN var=val cmd` should behave as described by [XCU Shell Command Language](http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_09_01). The value is exported only for cmd and not for "the current execution environment". In other words, `RUN var=val cmd` is nothing special to Docker, it's just a normal simple shell command.

I'm trying to walk the line and not explain basic UNIX to documentation readers, but at the same time, as much as possible make the documentation accessible for people who may not _completely_ understand basic UNIX.